src/lib/viewers/media/DashViewer.js

```diff
@@ -280,9 +301,9 @@ class DashViewer extends VideoBaseViewer {
      */
     handleAudioTrack() {
         const audioIdx = parseInt(this.cache.get('media-audiotracks'), 10);
-        if (this.audioTracks[audioIdx] !== undefined) {
-            const track = this.audioTracks[audioIdx];
-            this.player.selectAudioLanguage(track.language, track.role);
+        const newAudioTrack = this.audioTracks[audioIdx];
+        if (newAudioTrack !== undefined) {
+            this.enableAudioId(newAudioTrack.role);
```

Comment:
This'll throw an error if `newAudioTrack` is anything but an `Object`, due to accessing `role`, and only checking if it's not `undefined`

Reply:
this is how we build the audio tracks 
```
   this.audioTracks = uniqueAudioVariants.map((track) => ({
            language: track.language,
            role: track.roles[0]
        }));
```
We make this assumption that tracks coming back from the manifest are objects all over the code. Should I trust this as a fact or add the check everywhere else?